### PR TITLE
chore: update ruff linting scripts and settings

### PR DESCRIPTION
**Pull request by anakin87**
### Proposed Changes:

- I initially noticed that the script `hatch run lint:fmt` was not working -> I replaced `ruff --fix` with `ruff check --fix`.

- Then I started noticing some warnings like `warning: The top-level linter settings are deprecated in favour of their counterparts in the `lint` section...` -> I changed all of them in pyproject files.
- I also noticed that we introduced scripts like this: `style = ["ruff check {args:. --exclude tests/}",...]` in [recent PRs](https://github.com/deepset-ai/haystack-core-integrations/pulls?q=is%3Apr+is%3Aclosed+ruff).
  - I think it is preferable to specify exclusions in structured form in pyproject file (using `tool.ruff.exclude = ["tests"]`)
  - I would not exclude tests from linting and formatting.
  
  -> I removed these expressions and, only where necessary (too much work to adopt linting), replaced them with the appropriate settings in the pyproject file.

### How did you test it?
CI, local tests.

### Checklist

- I have read the [contributors guidelines](https://github.com/deepset-ai/haystack-core-integrations/blob/main/CONTRIBUTING.md) and the [code of conduct](https://github.com/deepset-ai/haystack-core-integrations/blob/main/CODE_OF_CONDUCT.md)
- I have updated the related issue with new insights and changes
- I added unit tests and updated the docstrings
- I've used one of the [conventional commit types](https://www.conventionalcommits.org/en/v1.0.0/) for my PR title: `fix:`, `feat:`, `build:`, `chore:`, `ci:`, `docs:`, `style:`, `refactor:`, `perf:`, `test:`.

### CODE REVIEW

File: .github/workflows/anthropic.yml [hunk 1]
```diff
@@ -15,7 +15,7 @@ defaults:
     working-directory: integrations/anthropic
 
 concurrency:
-  group: cohere-${{ github.head_ref }}
+  group: anthropic-${{ github.head_ref }}
```
Comment (anakin87) [inline .github/workflows/anthropic.yml:18]: unrelated: I noticed that Anthropic tests were not running due to the wrong concurrency group.

File: integrations/cohere/tests/test_cohere_chat_generator.py [hunk 2]
```diff
@@ -7,6 +7,7 @@
 from haystack.components.generators.utils import print_streaming_chunk
 from haystack.dataclasses import ChatMessage, ChatRole, StreamingChunk
 from haystack.utils import Secret
+
```
Comment (Amnah199) [inline integrations/cohere/tests/test_cohere_chat_generator.py:10]: Was this intentional? It added extra files in PR.

Comment (anakin87) [reply integrations/cohere/tests/test_cohere_chat_generator.py:10]: yeah, related to linting...

File: integrations/instructor_embedders/pyproject.toml [hunk 3]
```diff
@@ -152,13 +154,13 @@ unfixable = [
   "F401",
 ]
 
-[tool.ruff.isort]
+[tool.ruff.lint.isort]
 known-first-party = ["instructor_embedders"]
```
Comment (silvanocerza) [inline integrations/instructor_embedders/pyproject.toml:158]: Shouldn't we have `haystack_integrations` here too?

Comment (anakin87) [reply integrations/instructor_embedders/pyproject.toml]: :+1:

File: integrations/snowflake/tests/test_snowflake_table_retriever.py [hunk 4]
```diff
@@ -477,8 +477,10 @@ def test_to_dict_default(self, monkeypatch: MagicMock) -> None:
 
         data = component.to_dict()
 
+        t = "haystack_integrations.components.retrievers.snowflake.snowflake_table_retriever.SnowflakeTableRetriever"
+
         assert data == {
-            "type": "haystack_integrations.components.retrievers.snowflake.snowflake_table_retriever.SnowflakeTableRetriever",
+            "type": t,
```
[the remark this answers is not in the document]
Comment (anakin87) [reply integrations/snowflake/tests/test_snowflake_table_retriever.py]: done

**Review by silvanocerza — APPROVED** [review]
Niiiice! 🚀

**Pull request merged**